### ISSUE TITLE
(Bug?): getCellNode Not Returning Correct Rows in Latest Version (Regression from 8.1.5)

### ISSUE THREAD

**Issue by lechu90ns** [serenity-is/sleekgrid#12]
In the latest version of Serenity.Corelib (version 8.8.1), the method in grid.ts, of sleekgrid, getCellNode(row, cell) https://github.com/serenity-is/sleekgrid/blob/main/src/grid/grid.ts#L3827 does not return the expected DOM node for rows that are outside the currently rendered range. This issue was not present in version 8.1.5, where getCellNode reliably returned the requested rows regardless of their position in the grid.

**Steps to Reproduce**

1. Initialize a grid with a large dataset (enough to enable virtual scrolling).
2. Call getCellNode(row, cell) for a row that is currently not visible (e.g., scrolled out of view).
3. Observe that getCellNode returns null or an unexpected value.
Expected behavior in version 8.1.5:

- getCellNode(row, cell) returns the DOM node even if the row is not currently rendered.

Behavior in the latest version:

- getCellNode(row, cell) returns null for rows that are not rendered.

**Impact**
This issue impacts functionalities relying on dynamically interacting with rows that are not currently rendered (e.g., background updates, programmatically selecting rows, etc.). I use it to render printing for all data for reports (we cannot use it with pagination due to grouping summaries)

Screenshot from my local:
![image](https://github.com/user-attachments/assets/4ac69b8c-6edb-4744-b125-6010b36478b2)

**Workaround Attempted**
I've tried forcing the grid to render missing rows by manually entering `renderAllCells` and `renderAllRows` to true, but this is inefficient and causes performance issues, especially for large datasets.

**Additional Information**
This issue seems to be related to how the cache (_rowsCache) is maintained in the latest version. It appears that rows outside the visible range are no longer being cached, unlike in 8.1.5. However, I am not sure if that was the case.

**Request for Help**
Could you provide guidance on how to ensure that getCellNode(row, cell) works as expected for all rows, not just the ones currently rendered and having virtualization enabled at the same time?

Thank you for your time and support!

**Comment by volkanceylan** [comment]
getCellNode never returned cells outside view, maybe some buffer before after view. It is how virtualization works. Use renderAllRows and auto height for reports.

**Comment by lechu90ns** [comment]
My apologies, it appeared that I missed scrolling `_grid.scrollRowToTop(r);` while getting `_grid.getCellNode(r, c)` in slick.pring.js plugin. Anyway, Thank you! :)

**Comment by lechu90ns** [comment]
Ehh, I'm so sorry to bug you again, @volkanceylan. However, I just experienced this issue when I enabled virtualization.

![image](https://github.com/user-attachments/assets/2c9fb515-180e-4d82-8314-5c2d97de6e0e)

I used this code plugin, line 35 is bugging sleekGrid. 
https://github.com/wmluke/slickgrid-print-plugin/blob/master/src/scripts/slickgrid-print-plugin.js

It used to work with 8.1.5, is there anything that I can replace to make it work?
